File: app/src/main/java/com/murat/shporgalki/FactorialActivity.java
package com.murat.shporgalki;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.math.BigInteger;

public class FactorialActivity extends AppCompatActivity {
//    Обозачаю кнопки
    EditText number;
    TextView text;
//Стандарт
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme(R.style.Theme_Design_NoActionBar);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_factorial);
//        Текстовые поля
        number = (EditText) findViewById(R.id.Number);
        text = (TextView) findViewById(R.id.Text);
    }
//Назад
    public void onclickback(View view) {
        Intent back = new Intent(FactorialActivity.this, MathActivity.class);
        startActivity(back);

    }
//Нажатие на кнопку
    public void Onclickfind(View view) {
//       Если поле не пустое начинается поиск
        if (!number.getText().toString().equals("") && Integer.parseInt(number.getText().toString()) <= 20000) {
           //Факториал
           int n1 = Integer.parseInt(number.getText().toString());
           BigInteger result = BigInteger.ONE;
           for (int i = 1; i <= n1; i++) {
               result = result.multiply(BigInteger.valueOf(i));

           }
           text.setText(String.valueOf(result));
       }
        else  if (Integer.parseInt(number.getText().toString()) > 20000) {
            Toast.makeText(getApplicationContext(),"Числа больше 20001 не воспринимаются так как они слишком большие!",Toast.LENGTH_SHORT).show();
        }
       else
            Toast.makeText(getApplicationContext(),"Введите число!",Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/dvlopapps/shporgalki/FactorialActivity.kt
package com.dvlopapps.shporgalki

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import java.math.BigInteger

class FactorialActivity : AppCompatActivity() {
    //    Обозачаю кнопки
    var number: EditText? = null
    var text: TextView? = null

    //Стандарт
    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(R.style.Theme_Design_NoActionBar)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_factorial)
        //        Текстовые поля
        number = findViewById<View>(R.id.Number) as EditText
        text = findViewById<View>(R.id.Text) as TextView
    }

    //Назад
    fun onclickback(view: View?) {
        val back = Intent(this@FactorialActivity, MathActivity::class.java)
        startActivity(back)
    }

    //Нажатие на кнопку
    fun Onclickfind(view: View?) {
//       Если поле не пустое начинается поиск
        if (number!!.text.toString() != "" && number!!.text.toString().toInt() <= 20000) {
            //Факториал
            var n1 = number!!.text.toString().toInt()
            var result = BigInteger.ONE
            for (i in 1..n1) {
                result = result.multiply(BigInteger.valueOf(i.toLong()))
            }
            text!!.text = result.toString()
        } else if (number!!.text.toString().toInt() > 20000) {
            Toast.makeText(applicationContext, "Числа больше 20000 не воспринимаются так как они слишком большие!", Toast.LENGTH_SHORT).show()
        } else Toast.makeText(applicationContext, "Введите число!", Toast.LENGTH_SHORT).show()
    }
}